specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-07-01/virtualMachine.json

```diff
@@ -1891,6 +1891,72 @@
           "final-state-via": "location"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/attachDetachDataDisks": {
+      "post": {
```

Comment:
> Here, resource is Virtual machine and any operation on the virtual machine related to its modification should be classified under PATCH or PUT Http verb apart from virtual machine creation. I guess we should be going with Http PUT or PATCH verb rather than POST. POST looks a bit confusing basis REST semantics

Since this is an action on VM similar to start, restart, reimage, redeploy, etc, following POST HTTP verb. Please suggest.

Refer existing action APIs: https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/start?tabs=HTTP

Reply:
Since the older rest semantics under azure-rest-api-specs follows this, we are good to go with POST then.